### PR TITLE
CPU and Memory overuse when use push and the queue is over 100  or worker over 1k.

### PR DESCRIPTION
**Pull request by chensunny**
CPU and Memory overuse when use push and the queue is over 100  or worker
over 1k.

✅ Closes: https://github.com/bitleak/lmstfy/issues/56
✅ Closes: https://github.com/bitleak/lmstfy/issues/57

### CODE REVIEW

File: engine/redis/metrics.go [hunk 1]
```diff
@@ -53,7 +53,8 @@ const (
 func setupMetrics() {
 	cv := newCounterVecHelper
 	gv := newGaugeVecHelper
-	sv := newSummaryHelper
+	//sv := newSummaryHelper
```
Comment (git-hulk) [inline engine/redis/metrics.go:56]: remove `sv := newSummaryHelper`

File: push/meta.go [hunk 2]
```diff
@@ -267,8 +268,8 @@ func (mm *MetaManager) Dump() map[string]map[string][]string {
 }
 
 func (meta *Meta) Validate() error {
-	if meta.Workers <= 0 || meta.Workers > 16 {
-		return errors.New("workers should be between 1 and 16")
+	if meta.Workers <= 0 || meta.Workers > maxWorkerNum {
+		return errors.New(fmt.Sprintf("workers should be between 1 and %d",maxWorkerNum))
```
Comment (git-hulk) [inline push/meta.go:272]: code tidy =>
`fmt.Errorf("workers should be between 1 and %d", maxWorkerNum)`

**Review by git-hulk — CHANGES_REQUESTED** [review]

**Comment by liguangbo** [comment]
trvis的go版本应该升级到1.13。time.Milliseconds 是从1.13开始支持的，或者用Nanoseconds()/1e6 来代替。

File: server/handlers/metrics.go [hunk 3]
```diff
@@ -18,13 +18,13 @@ var metrics *PerformanceMetrics
...
-	latencies := prometheus.NewSummaryVec(
-		prometheus.SummaryOpts{
+	latencies := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
 			Namespace:  "infra",
 			Subsystem:  "lmstfy_http",
 			Name:       "latency_milliseconds",
 			Help:       "rest api latencies",
-			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.95: 0.001},
+            Buckets:   prometheus.ExponentialBuckets(15, 2.5, 9),
```
Comment (liguangbo) [inline server/handlers/metrics.go:27]: format code

**Review by liguangbo — CHANGES_REQUESTED** [review]

**Review by git-hulk — APPROVED** [review]
LGTM

**Review by liguangbo — APPROVED** [review]

**Pull request merged**